File: 0414-third-maximum-number/0414-third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        Set<Integer> new1=new HashSet<>();
        for(int num:nums){
            new1.add(num);
        }
        List<Integer> x=new ArrayList<>(new1);
        Collections.sort(x);
        
        int n=x.size();
        if(n>=3){
            return x.get(n-3);
        }
        return x.get(n-1);
    }
}

File: 0414-third-maximum-number/0414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        new1=set(nums)
        # print(new)
        new=list(sorted(new1))
        # print(new)
        n=len(new)
        if(n>=3):
            return new[n-3]
        else:
            return new[n-1]